src/modding/ModConfigItemInfo.cs

```
@@ -0,0 +1,214 @@
+                }
+
+                break;
+            case "bool":
+            case "boolean":
+            case "b":
+                var buttonUIElement = childUIElement as Button;
+                if (buttonUIElement != null)
+                {
+                    buttonUIElement.Pressed = Convert.ToBoolean(Value, CultureInfo.CurrentCulture);
+                }
+
+                break;
+            case "string":
+            case "s":
+                var stringUIElement = childUIElement as LineEdit;
+                if (stringUIElement != null)
+                {
+                    stringUIElement.Text = Convert.ToString(Value, CultureInfo.CurrentCulture);
+                }
+
+                break;
+            case "option":
+            case "enum":
+            case "o":
+                var optionUIElement = childUIElement as OptionButton;
+                if (optionUIElement != null)
+                {
+                    optionUIElement.Selected = Convert.ToInt32(Value, CultureInfo.CurrentCulture);
+                }
+
+                break;
+            case "color":
+            case "colour":
+            case "c":
+            case "alphacolor":
+            case "alphacolour":
+            case "ac":
+                var colorUIElement = childUIElement as ColorPickerButton;
+                if (colorUIElement != null && Value != null)
+                {
+                    colorUIElement.Color = new Color((string)Value);
+                }
+
+                break;
+        }
+    }
+
+    /// <summary>
+    ///   Returns a list of string based on the options the OptionButton can have
+    /// </summary>
+    public List<string> GetAllOptions()
+    {
+        var optionsJArray = Options as JArray;
+        if (optionsJArray == null)
+        {
+            return new List<string>();
+        }
+
+        return optionsJArray.ToObject<List<string>>()!;
+    }
```

I'm fairly sure there are several types of ConfigItemInfo where you don't want this to be called (I guess not all types have an `OptionButton`?). As such, it should raise an error when this happens, and not go silently with returning an empty list (I assume, please feel free to correct me if I'm wrong)

---

Yeah, since it should not be called if it not a option type

i changing to return a empty list and printerr?
does that sound fine?